libraries/AP_Camera/AP_RunCam.cpp

```diff
@@ -1058,8 +1069,86 @@ uint8_t AP_RunCam::Request::get_expected_response_length(const Command command)
     return 0;
 }
 
+// AP_Camera API
+
+// return true if healthy
+bool AP_RunCam::healthy() const
+{
+    return camera_ready();
+}
+
+// momentary switch to change camera between picture and video modes
+void AP_RunCam::cam_mode_toggle() {
+
+}
+
+// entry point to actually take a picture.  returns true on success
+bool AP_RunCam::trigger_pic() {
+    return false;
+}
+
+// send camera information message to GCS
+void AP_RunCam::send_camera_information(mavlink_channel_t chan) const
+{
+    // exit immediately if not initialised
+    if (!camera_ready()) {
+        return;
+    }
+
+    static const uint8_t vendor_name[32] = "RunCam";
+    uint8_t model_name[32] {};
+    strncpy((char *)model_name, _models[_cam_type.get()-1], MIN(sizeof(model_name), sizeof(_models[_cam_type.get()-1])));
+    const char cam_definition_uri[140] {};
+
+    // capability flags
+    uint32_t flags = 0;
+
+    if (has_feature(Feature::RCDEVICE_PROTOCOL_FEATURE_START_RECORDING)) {
+        flags = CAMERA_CAP_FLAGS_CAPTURE_VIDEO;
+    }
+
+    if (has_feature(Feature::RCDEVICE_PROTOCOL_FEATURE_CHANGE_MODE)) {
+        flags |= CAMERA_CAP_FLAGS_CAPTURE_IMAGE;
+    }
+
+    // send CAMERA_INFORMATION message
+    mavlink_msg_camera_information_send(
+        chan,
+        AP_HAL::millis(),       // time_boot_ms
+        vendor_name,            // vendor_name uint8_t[32]
+        model_name,             // model_name uint8_t[32]
+        0,                      // firmware version uint32_t
+        NaNf,                   // sensor_size_h float (mm)
+        NaNf,                   // sensor_size_v float (mm)
+        0,                      // sensor_size_v float (mm)
```

Comment:
bad comment?

Reply:
The comment is correct - we are just not filling in these fields